### PR TITLE
Use lerna for builds

### PR DESCRIPTION
**Pull request by johndayrichter**
Migrates the project to use the lerna build infrastructure, and makes a number of related changes to better represent the code as a Google-compliant monorepo:

- Creates a cached lerna build config
- Resets all package versions to (mostly arbitrary) version 0.5.0, so we have somewhere to go during the dev period.
- Set the standard documentation to version 0.5.0 to match the general version number reset.
- Correctly sets package.json repository values
- Sets package.json publishConfig values to use Google-internal npm publishing standards (go/npm-publish-service)
- Moves common tsconfig options to a single root file
- Switches from gts linter to eslint, since gts is incompatible with hoisted monorepos
- Moved to a single, shared lint config
- Moved to a single, shared .gitignore file at the top level
- Removed ./setup.sh (since lerna does the same thing more effectively)
- Updated README.md to document new lerna-based setup procedures

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "packages" : [
+    ".",
+    "opr-devtools",
+    "opr-models",
+    "opr-core",
+    "opr-sql-database",
+    "opr-google-cloud",
+    "opr-example-server"
+  ],
+  "version": "0.0.0"
```
Comment (ryckmanat) [inline lerna.json:12]: This feels a little odd - OK to have no version here? Do we need to start a formal versioning process with npm?

Comment (johndayrichter) [reply lerna.json:12]: According to the [lerna docs](https://lerna.js.org/docs/api-reference/configuration#version), the "version" property in lerna.json is used to record the last published version of your packages when you're publishing in fixed mode. I believe this starts as 0.0.0 when you first initialize a project and only should change when you actually publish the packages.

Comment (ryckmanat) [reply lerna.json:12]: Cool - sg!

**Review by ryckmanat — APPROVED** [review]
LGTM

**Pull request merged**